contrib/create-waku/src/cli.ts

```diff
@@ -0,0 +1,153 @@
+import { renderTemplate } from './renderTemplate'
+
+function isValidPackageName(projectName: string) {
+  return /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/.test(
+    projectName
+  )
+}
+
+function toValidPackageName(projectName: string) {
+  return projectName
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, '-')
+    .replace(/^[._]/, '')
+    .replace(/[^a-z0-9-~]+/g, '-')
+}
+
+// if the dir is empty or not exist
+function canSafelyOverwrite(dir: string) {
+  return !fs.existsSync(dir) || fs.readdirSync(dir).length === 0
+}
+
+async function init() {
+  const cwd = process.cwd()
+
+  let targetDir = '';
+  let defaultProjectName = 'waku-project'
+
+  const CHOICES = fs.readdirSync(path.resolve(cwd, 'template'))
+  let result: {
+    packageName?: string
+    shouldOverwrite?: string
+    chooseProject?: string
+  } = {}
+
+  try {
+    result = await prompts([
+      {
+        name: 'projectName',
+        type: 'text',
+        message: 'Project Name',
+        initial: defaultProjectName,
+        onState: (state: any) => (targetDir = String(state.value).trim() || defaultProjectName)
+      },
+      {
+        name: 'shouldOverwrite',
+        type: () => canSafelyOverwrite(targetDir) ? null : 'confirm',
+        message: `${targetDir} is not empty. Remove existing files and continue?`,
+      },
+      {
+        name: 'overwriteChecker',
+        type: (values: any) => {
+          if (values.shouldOverwrite === false) {
+            throw new Error(red('✖') + ' Operation cancelled')
+          }
+          return null
+        }
+      },
+      {
+        name: 'packageName',
+        type: () => (isValidPackageName(targetDir) ? null : 'text'),
+        message: 'Package name',
+        initial: () => toValidPackageName(targetDir),
+        validate: (dir: string) => isValidPackageName(dir) || 'Invalid package.json name'
+      },
+      {
+        name: 'chooseProject',
+        type: 'select',
+        message: 'Choose a starter template',
+        choices: [
+          { title: 'basic-template', value: CHOICES[0] },
+          { title: 'async-template', value: CHOICES[1] },
+          { title: 'promise-template', value: CHOICES[2] },
+        ],
+      }
+    ], {
+      onCancel: () => {
+        throw new Error(red('✖') + ' Operation cancelled')
+      }
+    })
+  } catch (cancelled) {
+    if(cancelled instanceof Error) 
+      console.log(cancelled.message)
+    process.exit(1)
+  }
+
+  const { packageName, shouldOverwrite, chooseProject } = result
+
+  const root = path.join(cwd, targetDir)
+
+  if (shouldOverwrite) {
+    emptyDir(root)
+  } else if (!fs.existsSync(root)) {
+    fs.mkdirSync(root)
+  }
+
+  const pkg = { name: packageName?? toValidPackageName(targetDir), version: '0.0.0' }
+
+  fs.writeFileSync(
+    path.resolve(root, 'package.json'),
+    JSON.stringify(pkg, null, 2),
+  )
+
+  console.log("Setting up project...")
+  const templateRoot = path.resolve(cwd, 'template')
```

Comment:
I think `path.resolve('template')` is the same. But, not super important anyway.